### PR TITLE
Ability to use basic search + filters at the same time

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/15371

This PR updates the project list search box and advanced filtering to work together so users can filter down filtered rows down in either direction. The logic is:
search box term (query with OR logic between search box fields shown in placeholder text)
AND 
advanced filters (query with AND or OR logic between rows of advanced filters)

## Testing
**URL to test:** 

https://deploy-preview-1263--atd-moped-main.netlify.app/moped/projects

**Steps to test:**
1. Go to the project list view and enter a search value like **testing** in the search box and press **Enter**. You should see filtered results. Note the number of results and the search params as you continue through the steps.
2. Open the advanced filters and add a filter like **Status is potential** and click **Search.** You should see the results that match the search box term and the advanced filter that you applied.
3. Add another filter row with a filter like **Lead contains coa**, and keep the any/all radios set to **Match all filters** You should see even less results based on the filters you chose. I see one match at this point with the examples that I suggested.
4. Set the any/all radios to **Match any filters** and click **Search**. You should see your search results include more rows.
5. Add another advanced filter like **Sponsor contains coa**. You should see more rows included again.
6. Remove filters one by one and click **Search** between each one to step back through your filters until only one advanced filter is applied.
7. **I could use some feedback on the UI flow in the test steps from here and on, thanks! 🙏**
8. You should see the trash can icon next to this last filter row. We used to hide it so the last filter could not be removed.
9. Remove the last filter and you should see the text **No filters applied.**. This allows users to click the **Search** button again to get results for only the remaining search box. The **Add filter** button can also be used here to get back to advanced filtering. Click the **Search** and you should search results that are only filtered by the term in the search box.
10. Click the advanced filters icon, and you should see a new row ready to accept an advanced filter.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

**Comment by mddilley** [comment]
@johnclary I'll demo where this is at in dev sync for more input, but I wanted to give you first dibs on review here to make sure this is fulfilling the requirements.

**Comment by johnclary** [comment]
@mddilley this is perfect! looks ready for review to me 👀

**Review by frankhereford — APPROVED** [review]
Mike, this is great. I noticed one thing that I think you may want to roll into this work, but it may be a feature and not a bug. Here's a video depicting searching for `John` and `John ` with a trailing space. 

https://github.com/cityofaustin/atd-moped/assets/3653206/28be448f-ffac-4686-bd46-5ea2840f2a26

My intuition was that those searches would be searching for the same thing, but that is just my take, so YMMV. In any case, it's not a regression, so 🚢 🚀 .

To your question about UI flow:

I do get that we want to the search button to be the thing that kicks off new searches, and I totally get that that means not to be searching as the user types with some debounced function or what-have-you, but I do think the clicking on a trash can and having a search term disappear from the UI warrants a new search to be performed. There's something about a click event that feels more like "do something" than a key-up event. 

The clicking on the trash-can is an event we can hook onto and keep the state of the results in sync with the visible configuration of the search, and my feeling is that we should, otherwise a user could hit that trashcan, get distracted by something, and then come back to the app and get real confused about why the advanced search configuration isn't aligned with the results. 

I do recognize that since we don't search on key-up, it's totally possible to get in this state where the configuration of the advanced search doesn't match the results by adding a new search term but not hitting the search button also ... but that process ends in a key-up event, and not a click event, so it feels more "OK" to me somehow. This is all opinion, so it's all YMMV, and I'm cool with this as it is also!

Thanks again!

File: moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useSearch.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const useSearch = ({ queryConfig }) => {
   const [searchTerm, setSearchTerm] = useState(simpleSearchValue ?? "");
 
   const searchWhereString = useMemo(() => {
-    if (searchTerm && searchTerm !== "") {
+    if (searchTerm && searchTerm.length > 0) {
```
[the remark this answers is not in the document]
Comment (johnclary) [reply moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useSearch.js:41]: i agree we ideally would trim leading and trailing whitespace. and also agree it makes sense to roll that into work related to the bug Mike mentions, and—best case scenario—as part of implementing a new default behavior that greedily searches to see if a project name contains any single word contained in the search string. 

we definitely need another issue written for those enhancements—will do!

**Review by johnclary — APPROVED** [review]
i find this to be a huge improvement over the current behavior. thank you @mddilley for stitching in this new behavior!!! 🚢🚢🚢🚢🚢🚢🚢

@frankhereford if i'm following you correctly, you're observing that there is a UI de-sync that can happen when you delete an advanced filter, because it immediately updates the URL search params, but you can close the filter dialog without hitting the **Search** button—so the search results on your page do not match the URL.

i agree that doesn't seem right, and i did just verify that behavior is not a regression. i will open a backlog issue to revisit the event handling here. i think the whole dialog could use a bit of design refinement as well 🙏

**Review by chiaberry — APPROVED** [review]
its exciting to see the behavior match up with what user expect to happen. I agree with John that the overall design could use some refinement. I think its a little weird that users can delete all the advanced filters, have an empty search bar and still click SEARCH, but that's on them imo.

**Comment by johnclary** [comment]
@mddilley i'm bringing thins into `main` 🚢 💨

**Pull request merged**